src/main/py/peer/peer.py

```diff
@@ -190,6 +190,47 @@ def merge_canvas(self, width: int, height: int, fragment) -> Image.Image:
 
         canvas.save("canvas.png", "PNG")
         return canvas
+    def add_to_art_collection(self, artwork, collection):
+        """
+        Add artwork to collection
+        """
+
+        try:
+            collection.add_to_art_collection(artwork)
+            logging.info("Artwork successfully added to collection.")
+        except ValueError as e:
+            logging.error("Failed to add artwork to collection: %s", e)
+
+    def remove_from_art_collection(self, artwork, collection):
+        """
+        Remove artwork from collection
+        """
+
+        try:
+            collection.remove_from_art_collection(artwork)
```

Comment:
Optional/Opportunistic Improvement

Python lists will only throw on `remove` if the item isn't in the list. It's better to avoid the scenario with a guard. Exception `throw`ing / `catch`ing is expensive, especially compared to a boolean check. If `artworks` is changed to `set` and `ArtCollection` made a `@dataclass` (so that `__eq__` is generated[1]), even better.

```python
if artwork in collection:
    collection.remove_from_art_collection(artwork)
```

[1] `__eq__` with value semantics is important for `set` and for an object that's going to be used as a key for a `map`/`dict`, since the default `__eq__` non-`@dataclass` classes get automatically has identity semantics.